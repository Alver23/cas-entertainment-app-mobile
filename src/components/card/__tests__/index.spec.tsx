// Dependencies
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { shallow, ShallowWrapper } from 'enzyme';

// Components
import { MemoizedCard } from '../index';

const MemoizedComponent = MemoizedCard.type;

describe('Card component', () => {
  let component: ShallowWrapper;

  const props = {
    backgroundImage: 'http://localhost/imgae.png',
    onPress: jest.fn(),
    text: 'fake text',
  };

  beforeEach(() => {
    component = shallow(<MemoizedComponent {...props} />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onPress void', () => {
    const onPress = component.find(TouchableOpacity).prop('onPress');

    onPress();
    expect(props.onPress).toHaveBeenCalled();
  });
});
