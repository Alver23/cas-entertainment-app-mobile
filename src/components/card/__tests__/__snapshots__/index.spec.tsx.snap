// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should save a snapshot of the component 1`] = `
<ForwardRef
  onPress={[MockFunction]}
  style={
    Array [
      Object {
        "borderRadius": 8,
        "borderWidth": 1,
        "flex": 1,
        "margin": 4,
        "padding": 8,
      },
      undefined,
    ]
  }
/>
`;
